llvm/tools/llvm-exegesis/lib/Error.h

```diff
@@ -41,14 +43,19 @@ class ClusteringError : public ErrorInfo<ClusteringError> {
 class SnippetCrash : public ErrorInfo<SnippetCrash> {
 public:
   static char ID;
-  SnippetCrash(const Twine &S) : Msg(S.str()) {}
+  SnippetCrash(const Twine &S)
```

Comment:
All right, sorry for the conflicting recommendations. It's hard to know exactly what the best approach is given that I can't see how you want to use it downstream. 

How are you accessing the signal and address if there are no getters for these ?

Reply:
Coming back to that patch to try to see whay I made that remark: I think that this was to avoid the `if (!E.isA<SnippetCrash>() && !E.isA<SnippetSegfaultCrash>())`. This is kind of related to https://github.com/llvm/llvm-project/pull/74211.

Reply:
Ah, makes sense.

Either way I think we would need a conditional there to send appropriate diagnostics to the user, but a generic separate error would be more scalable than individual error classes. I'm not sure there's a large benefit to separating out the error classes though, especially given it would end up looking pretty similar to D158052. What do you think is best in this situation?

Reply:
> https://github.com/llvm/llvm-project/pull/74211 adds additional support (including a getter). https://github.com/google/gematria/pull/30 shows the usage on top of that PR.

Thanks for the pointers. With the added getter in #74211 I think we should probably have two different types, because the getter would be meaningless in case of a failed child process without a crash. In particular, l137 in https://github.com/google/gematria/pull/30 would be incorrect. WDYT ?

Reply:
That's true. It makes the assumption that every crash is a segfault which is not correct. I'll refactor it to a separate error.